java/CMakeLists.txt

```diff
@@ -38,6 +38,19 @@ else()
   set(JAVA_PROFILE "-Pcmake,benchmark")
 endif()
 
+execute_process(
+  COMMAND java --add-opens java.base/java.nio=ALL-UNNAMED -version
+  WORKING_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}
+  RESULT_VARIABLE RET
+  OUTPUT_QUIET
+  ERROR_QUIET)
+if(RET EQUAL 0)
+  set(JAVA_PROFILE ${JAVA_PROFILE},java17)
+  set(JAVA_TEST_PROFILE -Pcmake,java17)
+  set(ADD_OPENS --add-opens)
+  set(JAVA_NIO java.base/java.nio=ALL-UNNAMED)
+endif()
```

Comment:
This is moved from [here](https://github.com/apache/orc/pull/911/files#diff-31ed790e942fadafe334a13765d2379b7b89b1c3d65a3973a94d55a77624c5dbL67-L77)